### PR TITLE
Apply iteration name provides to simple features as well

### LINKED ISSUE #1236: Iteration name provider does not apply to non parameterized tests in Spock 2
# Issue description

In Spock 1, the iteration name provider also applied to non parameterized tests. In Spock 2, it does not.

# How to reproduce

[spock-2-iteration-name-provider-bug.zip](https://github.com/spockframework/spock/files/5535315/spock-2-iteration-name-provider-bug.zip)

The above is a complete Gradle project that can run a test with Spock 1 or 2.

Works:

```
$ ./gradlew test -Pspock1

> Task :test

pkg.ExampleSpec > test [suffix] STARTED

pkg.ExampleSpec > parameterized [suffix] STARTED

pkg.ExampleSpec > parameterized [suffix] STARTED

```

Not works (Spock 2):

```
$ ./gradlew test

> Task :test

ExampleSpec > test STARTED

ExampleSpec > parameterized [suffix] STARTED

ExampleSpec > parameterized [suffix] STARTED
```

### PR DESCRIPTION
**Pull request by leonard84**
fixes #1236

### CODE REVIEW

File: spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/UnrolledFeatureMethods.groovy [hunk 1]
```diff
@@ -461,3 +494,26 @@ def "an actor (named #actor.getName()) age #actor.getAge()"() {
     }
   }
 }
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target([ElementType.TYPE])
+@Inherited
```
Comment (Vampire) [inline spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/UnrolledFeatureMethods.groovy:500]: This is just superfluous here, isn't it?

**Review by Vampire — CHANGES_REQUESTED** [review]
I'm not sure whether this is good and especially consistent.
The old way was also kind of a hack and it is questionable how and if it should be supported and if, how to do it consistently.
I extended your test example a bit, added `@Unroll`  to `parameterized` and added a `parameterized2` with no annotation for 1.3 and `@Rollup` for 2.0.
So we have:
`test` => simple feature
`parameterized` => unrolled feature (without the extension)
`parameterized2` => uprolled feature (without the extension)

With 1.3 I get this:
grafik

With 2.0 master I get this:
grafik

With this PR I get this:
grafik

So in 1.3
- it works on simple features name
- it does not work on already unrolled DD-iterations (DD = data-driven)
- it "works" on uprolled DD-iterations, unrolling them and defining the name, not appending

In 2.0 master
- it does not work on simple features name
- it does not work on DD-features name
- it works on already unrolled DD-iterations, appending
- it "works" on uprolled DD-iterations, unrolling them and defining the name, not appending

In this PR
- it does work on simple features name
- it does not work on DD-features name
- it works on already unrolled DD-iterations, appending
- it "works" on uprolled DD-iterations, unrolling them and defining the name, not appending

The original issue essentially was about difference between Spock 1 and 2, but with this PR it is still differente between the two and it is also inconsistent a bit in itself, for example it now works for the feature name of a simple feature but not for the feature name of a DD-feature.

File: spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/UnrolledFeatureMethods.groovy (hunk 1 above)
Comment (leonard84) [reply spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/UnrolledFeatureMethods.groovy:500]: yes it was c&p from the example code.

**Comment by leonard84** [comment]
Is it really inconsistent? The sample extension explicitly sets `feature.reportIterations = true` so that it negates the `@Rollup`. Just adding a constant suffix is probably not the real usecase so I can't really comment on that. Also there wouldn't have been a need to use `@Unroll` for 1.3 as the unrolling was already done with the `@Suffix` extension.

**Comment by Vampire** [comment]
The unrolled / uprolled is not so much the inconsistency but more showing that this is maybe more a hack than a proper feature.
The inconsistency is, that with this PR some feature names are affected (simple) and some not (data-driven).

**Comment by leonard84** [comment]
Well the simple feature also represents a single iteration, whereas the parametrized feature does not have its own iteration, as it is only a container for the individual iterations.

**Comment by Vampire** [comment]
That's about technical details and then it depends on the exact terminology used.
In Spock terms this is correct, in JUnit platform terms it is not as the container is also reported as test. ;-)

But I'm talking less about the technical correctness but more about the perceived correctness as seen by the user.
@ldaley as you reported the original issue, what is your opinion?

**Comment by ldaley** [comment]
My usage of this in Spock 1 was a “hack” insofar as it was abusing this mechanism in order to augment test names with global parameterisation context. 

We have a system where we parameterise entire test suites via system properties when launching, and desire to have the parameters be apparent in the reported test names. It's a very important aspect of our Spock usage.

Without understanding all of the details, I can see that a mechanism more focussed on augmenting the “reported test name” would be better than relying on this indirect mechanism. I don't think it would be a problem if that mechanism is different between 1 and 2.

**Comment by leonard84** [comment]
@ldaley take another look (`UnrolledFeatureMethods`), you can now properly set the `displayName` of a feature via an extension, which you also could do in Spock 1.x, but that affected iteration naming as well.

**Comment by ldaley** [comment]
@leonard84 LGTM. Looking forward to using it.

File: spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/UnrolledFeatureMethods.groovy (hunk 1 above)
Comment (szpak) [reply spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/UnrolledFeatureMethods.groovy:500]: So, maybe it could be removed?

**Review by szpak — APPROVED** [review]

**Pull request merged**